tests/integration-tests/src/test/java/org/apache/activemq/artemis/tests/integration/cluster/expiry/ClusteredExpiryTest.java

```diff
@@ -0,0 +1,115 @@
+      servers[0].getAddressSettingsRepository().addMatch("queues#", new AddressSettings().setExpiryQueueSuffix(SimpleString.of(".Expiry")).setAutoCreateExpiryResources(true).setExpiryAddress(SimpleString.of("Expiry")));
+      servers[0].getAddressSettingsRepository().addMatch("$#", new AddressSettings().setExpiryQueueSuffix(SimpleString.of(".SNFExpiry")).setAutoCreateExpiryResources(true).setExpiryAddress(SimpleString.of("SNFExpiry")));
+
+      servers[1].getAddressSettingsRepository().clear();
+      servers[1].getAddressSettingsRepository().addMatch("queues#", new AddressSettings().setExpiryQueueSuffix(SimpleString.of(".Expiry")).setAutoCreateExpiryResources(true).setExpiryAddress(SimpleString.of("Expiry")));
+      servers[1].getAddressSettingsRepository().addMatch("$#", new AddressSettings().setExpiryQueueSuffix(SimpleString.of(".SNFExpiry")).setAutoCreateExpiryResources(true).setExpiryAddress(SimpleString.of("SNFExpiry")));
+
+      Queue serverQueue0 = servers[0].createQueue(QueueConfiguration.of("queues." + getName()).setDurable(true).setRoutingType(RoutingType.ANYCAST));
+      servers[1].createQueue(QueueConfiguration.of("queues." + getName()).setDurable(true).setRoutingType(RoutingType.ANYCAST));
+      servers[0].createQueue(QueueConfiguration.of("Expiry" + getName()).setDurable(true).setRoutingType(RoutingType.ANYCAST));
+      servers[1].createQueue(QueueConfiguration.of("Expiry." + getName()).setDurable(true).setRoutingType(RoutingType.ANYCAST));
+
+      waitForBindings(0, "queues." + getName(), 1, 0, true);
+      waitForBindings(1, "queues." + getName(), 1, 0, true);
+
+      waitForBindings(0, "queues." + getName(), 1, 0, false);
+      waitForBindings(1, "queues." + getName(), 1, 0, false);
+
+      // pausing the SNF queue to keep messages stuck on the queue
+      servers[0].getPostOffice().getAllBindings().filter(f -> f.getUniqueName().toString().startsWith("$.artemis.internal.sf")).forEach(this::pauseQueue);
+      assertNotNull(snfPaused);
+
+      long NUMBER_OF_MESSAGES = 100;
+
+      ConnectionFactory factory = CFUtil.createConnectionFactory("CORE", "tcp://localhost:61616");
+      ConnectionFactory factory2 = CFUtil.createConnectionFactory("CORE", "tcp://localhost:61617");
+      try (Connection connection = factory.createConnection()) {
+         Session session1 = connection.createSession(true, Session.SESSION_TRANSACTED);
+         MessageProducer producer = session1.createProducer(session1.createQueue("queues." + getName()));
+         producer.setTimeToLive(2_000);
+         for (int i = 0; i < NUMBER_OF_MESSAGES; i++) {
+            producer.send(session1.createTextMessage("hello"));
+         }
+         session1.commit();
+      }
+      Wait.assertEquals(0L, serverQueue0::getMessageCount, 50_000, 100);
+      Wait.assertEquals(0L, snfPaused::getMessageCount, 50_000, 100);
+      Queue expiryQueue = servers[0].locateQueue("Expiry", "EXP.queues." + getName() + ".Expiry");
```

Comment:
It would be a lot easier to follow this test if there were variables for the prefix, suffix, and 'main queue name' values that were then used to set up the configuration and composed for the other usages such as this (maybe via another variable).  Especially given the slightly overloaded use of "Expiry" in many of them.